test/phoenix/controller/flash_test.exs

```diff
@@ -122,4 +123,37 @@ defmodule Phoenix.Controller.FlashTest do
       conn(:get, "/") |> fetch_flash([])
     end
   end
+
+  test "fetch based on default liveview cookie" do
+    flash = Flash.sign_token(FlashEndpoint, "liveview_salt456", %{notice: "hi"})
+    conn = conn(flash_cookie: flash) |> Flash.call([])
+    assert get_flash(conn, :notice) == "hi"
+    assert get_flash(conn, "notice") == "hi"
+  end
+
+  test "fetch based on provided cookie" do
+    flash = Flash.sign_token(FlashEndpoint, "othercookie_salt456", %{info: "hello"})
+    conn = conn(flash_cookie: flash) |> Flash.call(signing_salt: "othercookie_salt456")
+    assert get_flash(conn, :info) == "hello"
+    assert get_flash(conn, "info") == "hello"
+  end
```

Comment:
Two tests to show that if you have the same salt, then you can use the Flash plug to get your answer.